client/src/SubscriptionSelection.spec.js

```diff
@@ -0,0 +1,33 @@
+import { render, screen } from "@testing-library/react";
+import userEvent from "@testing-library/user-event";
+import SubscriptionSelection from "./SubscriptionSelection.js";
+
+describe("<SubscriptionSelection />", () => {
+  it("renders correctly", () => {
+    const component = render(<SubscriptionSelection onClick={() => {}} />);
+    expect(component.asFragment()).toMatchSnapshot();
+  });
+
+  it("renders a subscription button", () => {
+    render(<SubscriptionSelection />);
+    expect(
+      screen.getByRole("button", { name: "submit a subscription type" }),
+    ).toBeVisible();
+  });
+  it("renders the text", () => {
+    render(<SubscriptionSelection />);
+    expect(screen.getByText("Select a Subscription Type")).toBeVisible();
+  });
```

Comment:
Hi Ange, i think it could help to be more explicit with the test name here, this is a suggestion:

```suggestion
  it("renders the section header", () => {
    render(<SubscriptionSelection />);
    expect(screen.getByText("Select a Subscription Type")).toBeVisible();
  });
```

or you could even keep it more generic and just have a test that says `SubscriptionSelection renders correctly` and include the expectations for it rendering correctly inside of it (`displays continue button`, `displays header`...).

I added this suggestion because it think it's good to clarify what text you're asserting on in the page. Unit tests can even double as a form of documentation to help understand a component 📖

Reply:
Ok, this text was asserting on the text within the button. What we ended up doing was moving the assertion to another it block and removing the 'it' block "renders the section header". This helped to simplify the tests to be more aligned with the idea of testing as documentation.